File: Search In a Rotated Array/Leetcode_33.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if not nums: return -1
        lo, hi, t = 0, len(nums)-1, (target<nums[0], target)
        while lo <= hi:
            mid = (lo+hi)//2
            guess = (nums[mid] < nums[0], nums[mid])
            if guess == t: return mid
            elif guess < t: lo=mid+1
            else: hi=mid-1
        return -1

File: Search In a Rotated Array/Leetcode_33.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        if (nums.size()==0) return -1;
        int lo=0, hi=nums.size()-1, mid;
        while (lo<=hi){
            mid = lo + (hi-lo)/2;
            cout<<mid<<" "<<nums[mid]<<endl;
            if (nums[mid] == target) return mid;
            if (((nums[0]<nums[mid]) && (nums[0]>target)) || (nums[mid] < target)) {
                lo = mid+1;
            } else hi = mid-1;
        }
        return -1;
    }
};
